lib/main.dart

```diff
@@ -71,14 +71,7 @@ Future<void> _impl(List<String> args) async {
     channel = sdk;
     version = raw ? 'latest' : (await latestPublishedVersion(channel, flavor));
   } else if (sdk == 'main') {
-    // Check for `main` first and fall back to `be`. This handles the channel
```

Comment:
We'll want a changelog entry for this, and will need to bump the action's version as well.

Thinking through how user's will see this change:

- old versions of the action, testing against old sdks, will work
- new versions of the action (after this change), testing against old sdks (2.19, 3.0, ...) will fail
- old versions of the action, testing against newer sdks, will fail
- new versions of the action, testing against new sdks, will work

Is this change essentially dropping support for sdks before some version? Or, just for `main` channel sdks before some version? We probably don't need to support many previous versions for those sdks.

Can you update the changelog w/ an entry, and in that entry mention for which sdk version we first started writing artifacts into the `main` bucket?

Reply:
The main channel was only ever 'supported' for latest. There is no support in here for pinning a particular hash or an older SDK version.

Anyone that uses the old action before 1.5.0 (or what it was) will now be stuck on that last be release as of today, and anyone on that version or newer will use the main channel as of today (the compat logic I'm removing here automatically switched them together).

I'll add some changelog stuff and bump the version, this was just to get started while it was on my mind :)

Note that I am starting to feel we really should drop support for the raw flavor. The main channel builds is not a supported product in any way and the binaries aren't signed. I may even decide to store these artifacts elsewhere in the future to guard the release bucket even more closely. I'll also want to discontinue the whole raw/signed flavor concept, since people should not really be poking in there and I need to change the layout for unrelated reasons, and it'll massively simplify this action.  But that's a discussion for another day.

Reply:
Gotcha, thanks for the clarification re: how breaking this is; ~sounds like this will not be a breaking change. I think either a match version rev or a minor version would be appropriate.

> Note that I am starting to feel we really should drop support for the raw flavor.

I think the simplifications to this action would be welcome. We probably want to open an issue on this repo in order to discuss, get feedback, and help any user that might be impacted plan alternatives.